### PR TITLE
OPSEXP-2755 Use alfresco-build-tools/maven-deploy-file action to publish release on nexus

### PR DESCRIPTION
**Pull request by pmacius**
Ref: OPSEXP-2755

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -39,12 +39,14 @@ jobs:
...
-          -F "v=${VERSION}" -F "p=zip" -F "file=@alfresco-ansible-deployment-${VERSION}.zip"
-          -u "$NEXUS_USERNAME:$NEXUS_PASSWORD"
-          https://artifacts.alfresco.com/nexus/service/local/artifact/maven/content
+        with:
+          group-id: org.alfresco
+          artifact-id: alfresco-ansible-deployment
+          repository-url: https://nexus.alfresco.com/nexus/content/repositories/releases/
+          version: ${{ env.VERSION }}
+          file: alfresco-ansible-deployment-${{ env.VERSION }}.zip
+          classifier: zip
```
Comment (pmacius) [inline .github/workflows/release.yml:50]:
```suggestion
```
Not sure if needed

Comment (gionn) [reply .github/workflows/release.yml:50]: I guess it's what `-F "p=zip"` was doing so would keep it for retrocompat

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -39,12 +39,14 @@ jobs:
...
-          NEXUS_PASSWORD: ${{ secrets.nexus_password }}
-        run: >-
-          curl -F "r=releases" -F "g=org.alfresco" -F "a=alfresco-ansible-deployment"
-          -F "v=${VERSION}" -F "p=zip" -F "file=@alfresco-ansible-deployment-${VERSION}.zip"
-          -u "$NEXUS_USERNAME:$NEXUS_PASSWORD"
-          https://artifacts.alfresco.com/nexus/service/local/artifact/maven/content
+        with:
+          group-id: org.alfresco
+          artifact-id: alfresco-ansible-deployment
+          repository-url: https://nexus.alfresco.com/nexus/content/repositories/releases/
```
Comment (gionn) [inline .github/workflows/release.yml:47]: how did you end up with this url?

Comment (alxgomz) [reply .github/workflows/release.yml:47]: looks ok to me as it's the primary URL of the hosted repo where existing artifacts are stored

Comment (pmacius) [reply .github/workflows/release.yml:47]: we use this for pfd-renderer https://github.com/Alfresco/alfresco-pdf-renderer/blob/master/.github/workflows/build-all.yml#L154

**Review by gionn — APPROVED** [review]

**Pull request merged**